sys/net/gnrc/network_layer/sixlowpan/frag/sfr/gnrc_sixlowpan_frag_sfr.c

```diff
@@ -1547,33 +1667,34 @@ static void _send_ack(gnrc_netif_t *netif, const uint8_t *dst, uint8_t dst_len,
           gnrc_netif_addr_to_str(dst, dst_len, addr_str),
           hdr->tag, bitmap[0], bitmap[1], bitmap[2], bitmap[3]);
     if (ack != NULL) {
-        _send_frame(ack, NULL, 0);
+        _send_frame(ack, NULL, NULL, 0);
     }
     else {
         DEBUG("6lo sfr: unable to build ACK for sending\n");
     }
 }
 
-static void _sched_next_frame(void)
+static void _sched_next_frame(gnrc_sixlowpan_frag_fb_t *fbuf)
 {
-    if (CONFIG_GNRC_SIXLOWPAN_SFR_INTER_FRAME_GAP_US > 0) {
+    if (gnrc_sixlowpan_frag_sfr_congure_snd_has_inter_frame_gap()) {
         int state = irq_disable();  /* make timer check atomic */
         bool already_set = xtimer_is_set(&_if_gap_timer);
 
         irq_restore(state);
         if (!already_set) {
             uint32_t last_sent_since = (_last_frame_sent - xtimer_now_usec());
+            uint32_t if_gap = gnrc_sixlowpan_frag_sfr_congure_snd_inter_frame_gap(fbuf);
 
-            if (last_sent_since <= CONFIG_GNRC_SIXLOWPAN_SFR_INTER_FRAME_GAP_US) {
-                uint32_t offset = CONFIG_GNRC_SIXLOWPAN_SFR_INTER_FRAME_GAP_US -
-                                  last_sent_since;
+            if (last_sent_since <= if_gap) {
+                uint32_t offset = if_gap - last_sent_since;
                 DEBUG("6lo sfr: arming inter-frame timer in %" PRIu32 " us\n",
                       last_sent_since);
+                _if_gap_msg.content.ptr = fbuf;
                 xtimer_set_msg(&_if_gap_timer, offset, &_if_gap_msg, _getpid());
             }
             else {
                 DEBUG("6lo sfr: send frame immediately\n");
-                gnrc_sixlowpan_frag_sfr_inter_frame_gap();
+                gnrc_sixlowpan_frag_sfr_inter_frame_gap(fbuf);
```

Comment:
Won't this cause an infinite recursion with `gnrc_sixlowpan_frag_sfr_inter_frame_gap()` calling `_sched_next_frame()`?